tests/pipeline/PipelineCTracing.cpp

```diff
@@ -0,0 +1,109 @@
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/PipelineC/PipelineC.h"
+#include "revng/PipelineC/Tracing/Private.h"
+#include "revng/PipelineC/Tracing/Trace.h"
+#include "revng/Support/TemporaryFile.h"
+
+#define BOOST_TEST_MODULE PipelineCTracing
+bool init_unit_test();
+#include "boost/test/unit_test.hpp"
+
+#include "revng/UnitTestHelpers/UnitTestHelpers.h"
+
+namespace rt = revng::tracing;
+
+class TheHarness {
+public:
+  std::string Buffer;
+  std::optional<rt::Trace> Trace;
+
+public:
+  bool parseTrace() {
+    auto MemoryBuffer = llvm::MemoryBuffer::getMemBuffer(Buffer);
+    auto MaybeTrace = rt::Trace::fromBuffer(*MemoryBuffer);
+    if (!MaybeTrace) {
+      llvm::consumeError(MaybeTrace.takeError());
+      return false;
+    }
+    Trace = MaybeTrace.get();
+    return true;
+  }
+};
+
+static TheHarness *Harness;
+
+struct Fixture {
+public:
+  Fixture() {
+    using ArgvT = std::vector<const char *>;
+    ArgvT Argv = { "",
+                   "-load=lib/librevngStringContainerLibrary.so",
+                   "-pipeline-path=tests/pipeline/Tracing.yml" };
+    rp_initialize(Argv.size(), Argv.data(), 0, {});
+
+    Harness = new TheHarness();
+  }
+
+  ~Fixture() {
+    delete Harness;
+    rp_shutdown();
+  }
+};
+
+BOOST_AUTO_TEST_SUITE(PipelineCTracingTestSuite,
+                      *boost::unit_test::fixture<Fixture>())
```

Comment:
Using these will make our life more annoying when we'll migrate away.
Do you really need it?

Consider that each test case must be executable on its own (there command line arguments to run individual tests), and the current configuration does not seem to support this.

Having multiple tests makes sense to start with something simple and go more complex, but there can be no information sharing between one test case and another (in particular not through a global variable that is allocated on the heap through naked `new`!).

Reply:
Reworked the test. Kept the fixture for `rp_initialize` and `rp_shutdown` though